components/net/hsts.rs

```diff
@@ -139,15 +142,160 @@ impl HstsList {
     }
 
     /// Step 10 of https://fetch.spec.whatwg.org/#concept-main-fetch.
-    pub fn switch_known_hsts_host_domain_url_to_https(&self, url: &mut ServoUrl) {
-        if url.scheme() != "http" {
+    pub fn apply_hsts_rules(&self, url: &mut ServoUrl) {
+        if url.scheme() != "http" && url.scheme() != "ws" {
             return;
         }
-        if url
-            .domain()
-            .map_or(false, |domain| self.is_host_secure(domain))
+
+        let mut upgrade_scheme = false;
```

Comment:
Nit: can we make this not `mut` by assigning a boolean expression?

Reply:
Should I change the other occurrences of `upgrade_scheme` to `let upgrade_scheme` instead?

Reply:
No, just initialize `upgrade_scheme` to be the "right" value, rather than initializing it to be false then updating it.